src/Microsoft.AspNetCore.Authentication/ClaimsTransformationMiddleware.cs

```diff
@@ -35,17 +36,22 @@ public class ClaimsTransformationMiddleware
 
         public async Task Invoke(HttpContext context)
         {
-            var handler = new ClaimsTransformationHandler(Options.Transformer, context);
+            var transform = Options.Transformer;
+            if (transform == null && Options.TransformerType != null)
```

Comment:
Yeah so what we have today is geared towards a single claims transformer instance, this PR makes it a bit easier to have multiple claims transformation middlewares, and also pull the transform from DI (needed basically only for scoped transformers)